core/src/replay_stage.rs

```diff
@@ -2188,9 +2193,8 @@ impl ReplayStage {
         if last_voted_slot < heaviest_bank_on_same_fork.slot()
             && !heaviest_bank_on_same_fork.slot_within_slothash(&last_voted_slot)
         {
-            let new_root = tower.record_bank_vote(heaviest_bank_on_same_fork, vote_account_pubkey);
-            // Because the last vote is too old, new_root should be none.
-            assert!(new_root.is_none());
+            *last_vote_unable_to_land = true;
```

Comment:
If we moved `refresh_vote()` to be after `select_vote_and_reset_forks()` could we then move this `last_vote_unable_to_land` entirely inside `select_vote_and_reset_forks()` so that we don't have to pass it around?

Reply:
I was thinking about this. So, if you voted on something new in select_vote_and_reset_forks, then you don't really need a vote refresh any more. That might be a better approach, but the behavior is also different from now.

Reply:
It's true, but I don't think it's a big change in behavior